jobwatcher/plugins/utils.py

```diff
@@ -0,0 +1,56 @@
+import shlex
+import subprocess as sub
+import os
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def run_command(command, env):
+    _command = shlex.split(command)
+    try:
+        DEV_NULL = open(os.devnull, "rb")
+        env.update(os.environ.copy())
+        process = sub.Popen(_command, env=env, stdout=sub.PIPE, stderr=sub.STDOUT, stdin=DEV_NULL)
+        _output = process.communicate()[0]
+        return _output
+    except sub.CalledProcessError:
+        log.error("Failed to run %s\n" % _command)
+        exit(1)
+    finally:
+        DEV_NULL.close()
+
+
+def get_optimal_nodes(nodes_requested, slots_requested, instance_properties):
+    vcpus = instance_properties.get('slots')
+    slots_remaining_per_node = []
+
+    for node_idx, node in enumerate(nodes_requested):
+        log.info("Requested node %s with slots %s" % (node, slots_requested[node_idx]))
+        # for simplicity, uniformly distribute the numbers of cpus requested across all the requested nodes
+        slots_required_per_node = -(-slots_requested[node_idx] // node)
+
+        if slots_required_per_node > vcpus:
```

Comment:
`slots_required_per_node` is the result of the total number of required slots (cpus for us) of a job divided by the number of requested nodes for the same job. If this is greater than the capacity of our instance (`vcpus`) then the job simply will not be submitted and will remain pending in Slurm. Today this causes a continuous scale up till the max, with this patch we should raise the issue in the log and avoid to scale for those unsatisfiable jobs.